custom_components/plejd/pyplejd/mesh.py

```diff
@@ -28,7 +30,8 @@ def __init__(self):
 
         self._ble_lock = asyncio.Lock()
 
-    def add_mesh_node(self, device):
+    def add_mesh_node(self, ble_device: BLEDevice, rssi: int):
+        device = BLEDeviceWithRssi(ble_device, rssi)
         if device not in self.mesh_nodes:
```

Comment:
This check will always fail so there will be a risk of duplicate devices in the node list.